### DISCUSSION TITLE
"Core-shell" composite active material

### DISCUSSION THREAD

**Discussion by lefile** [pybamm-team/PyBaMM#3615] [Ideas]
Hello!

There are electrode designs where one active material is coated on the surface of a second active material particle. It is effectively a composite electrode, but not in a way implemented right now, where composite active materials are independent. This could be implemented as a one particle with radially-dependent solid diffusion properties and similar.

**Comment by TomTranter** [comment]
Spatially dependent diffusion would be simple enough but I'd worry that the continuum representation wouldn't be enough to model things like grain boundary and stresses. Any FE models out there or data looking into this type of thing?

**Comment by rtimms** [comment]
There is [this](https://www.sciencedirect.com/science/article/pii/S0378775322014380) core-shell model for positive electrode degradation. The authors implemented it in their own fork of PyBaMM (link in paper), so you could probably take a similar approach to handle a coated particle (thinking carefully about what happens at the interface).

**Reply by rtimms** [reply]
@mzzhuo are there any plans to merge this into PyBaMM?

**Reply by mzzhuo** [reply]
Hi Robert, recently I got an email from someone at BMW asking about our PE phase-transition degradation submodel. I'd be thrilled to see it included in the official PyBaMM release. Do you know if the PyBaMM team is on board with that? If so, I might need your guidance on how to proceed. It's based on an older PyBaMM version, and it would also be great if you can help to review the code for efficiency and consistency with the current source code.

**Reply by rtimms** [reply]
We'd love to see that happen. The first step would be to bring your fork up to date with the latest `develop` branch, then you will be able to make a pull request and we can review it.

**Reply by TomTranter** [reply]
Nice paper @mzzhuo - just seeing for the first time. I would be interested in helping review any PR that you could create.

**Reply by mzzhuo** [reply]
Thanks Tom and Rob, I will try to update the branch and make a PR

**Reply by mzzhuo** [reply]
Hi Tom and Rob, since Pybamm has changed so much, updating my previous branch did not work well, so I just build the submodel upon the latest Pybamm release, now it is almost done and I wrote a script tutorial that can reproduce the results in my paper. I am now testing the codes (nox -s unit), but it will take some time and may not pass as I defined some domains, expression tree etc. At the same time, would the pybamm team like to review the source codes to speed up the whole process?

**Reply by rtimms** [reply]
Great! Yeah, if you like some feedback you can open a pull request and mark it as draft until it’s ready for a proper review.

**Reply by mzzhuo** [reply]
Hi Rob, I have created a PR, but it may not look proper like others, as it is my first time to contribute. At the moment, I think it would good if the pybamm team can seriously check/review the physics including equations, boundary conditions, etc in the submodel. Also, I kind of found two pybamm bugs (one leads to crash one to computation inaccuracy) which need attention. Here is a [link](https://hkustconnect-my.sharepoint.com/:b:/g/personal/mzhuo_connect_ust_hk/EYmFXnyOpP9KlUsrHg_0RtIBqQoIkN6qIpVCKu262uSgLA?e=Jtp6qP)  to my handwritten formula of scaling equations and bcs for your reference.

**Reply by rtimms** [reply]
Thanks, will take a look. Can you open separate issues for the bugs please?

**Reply by mzzhuo** [reply]
Sure, I will raise the issues after the pe submodel is merged as it may be invoked for demonstration
